### DISCUSSION TITLE
Grafana happily shows times from The Future

### DISCUSSION THREAD

**Discussion by andyross** [teslamate-org/teslamate#1750] [Ideas]
Click on the "zoom out" icon at the top of the grafana page a few times to see more data.

Note that it zooms out symmetrically, expanding the right and left boundaries of the data set by X%.

Note that this means the right edge of the data is now showing times that are in the future.

Clearly the desired behavior is to clamp to valid time ranges.

**Comment by adriankumpf** [comment]
Did I understand you correctly that you'd like TelsaMate to change the way Grafana works? 😉 It's not possible to customize the zooming functionality afaik.

**Comment by andyross** [comment]
Grafana has hooks for adding code, yes.  I don't know anything about it but it's absolutely something I've seen people do...

Did I understand you correctly that you view the Grafana integration in Teslamate as not something you intend to improve?  The grafana output is the only reason I'm interested...

**Comment by adriankumpf** [comment]
I'll move this to the feature requests section here: [💡 Ideas](https://github.com/adriankumpf/teslamate/discussions/categories/ideas) . If anyone has any ideas on how to implement this, let us know.
